Debug information for file: javali_tests/othertests/A3_team/FuncCallNull.javali
AST Resulting From Parsing Stage:
ClassDecl (name: Main, superClass: Object)
| MethodDecl (argumentNames: [], argumentTypes: [], name: main, returnType: void)
| | Seq ()
| | | VarDecl (name: a1, type: A1)
| | Seq ()
| | | Assign ()
| | | | Var (name: a1)
| | | | NewObject (typeName: A1)
| | | MethodCall ()
| | | | MethodCallExpr (methodName: foo)
| | | | | Var (name: a1)
| | | | | NullConst ()
ClassDecl (name: A1, superClass: Object)
| MethodDecl (argumentNames: [x], argumentTypes: [A], name: foo, returnType: void)
| | Seq ()
| | Seq ()
ClassDecl (name: A, superClass: Object)

Test failed because an exception was thrown:
    null
Stack trace:
java.lang.NullPointerException
	at cd.frontend.semantic.TableCreator.methodCall(TableCreator.java:529)
	at cd.frontend.semantic.TableCreator.methodCall(TableCreator.java:1)
	at cd.ir.Ast$MethodCallExpr.accept(Ast.java:460)
	at cd.ir.Ast$Expr.accept(Ast.java:85)
	at cd.ir.AstVisitor.visit(AstVisitor.java:16)
	at cd.ir.AstVisitor.visitChildren(AstVisitor.java:28)
	at cd.ir.AstVisitor.dflt(AstVisitor.java:37)
	at cd.ir.AstVisitor.dfltStmt(AstVisitor.java:43)
	at cd.ir.AstVisitor.methodCall(AstVisitor.java:92)
	at cd.ir.Ast$MethodCall.accept(Ast.java:602)
	at cd.ir.AstVisitor.visit(AstVisitor.java:16)
	at cd.ir.AstVisitor.visitChildren(AstVisitor.java:28)
	at cd.ir.AstVisitor.dflt(AstVisitor.java:37)
	at cd.ir.AstVisitor.seq(AstVisitor.java:100)
	at cd.ir.Ast$Seq.accept(Ast.java:677)
	at cd.ir.AstVisitor.visit(AstVisitor.java:16)
	at cd.frontend.semantic.TableCreator.methodDecl(TableCreator.java:119)
	at cd.frontend.semantic.TableCreator.classDecl(TableCreator.java:62)
	at cd.frontend.semantic.SemanticAnalyzer.check(SemanticAnalyzer.java:168)
	at cd.Main.semanticCheck(Main.java:111)
	at cd.test.AbstractTestAgainstFrozenReference.testSemanticAnalyzer(AbstractTestAgainstFrozenReference.java:146)
	at cd.test.AbstractTestAgainstFrozenReference.test(AbstractTestAgainstFrozenReference.java:50)
	at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:745)
